### ISSUE TITLE
Error in if (grepl(pattern, makefile_txt[i])) { :  argument is of length zero

### ISSUE THREAD

**Issue by XiangyunHuang** [stan-dev/rstan#634]
#### Summary:
When I run `stan` function from rstan, it gives me errors like this 

```
Error in if (grepl(pattern, makefile_txt[i])) { : 
  argument is of length zero
```

#### Description:

My Code is right, because I get correct results under  ubuntu 18.04.2

#### Reproducible Steps:

```
# modified code from https://github.com/stan-dev/example-models/tree/master/bugs_examples/vol1/rats
N <- 30
T <- 5
y <- structure(c(
  151, 145, 147, 155, 135, 159, 141, 159, 177, 134,
  160, 143, 154, 171, 163, 160, 142, 156, 157, 152, 154, 139, 146,
  157, 132, 160, 169, 157, 137, 153, 199, 199, 214, 200, 188, 210,
  189, 201, 236, 182, 208, 188, 200, 221, 216, 207, 187, 203, 212,
  203, 205, 190, 191, 211, 185, 207, 216, 205, 180, 200, 246, 249,
  263, 237, 230, 252, 231, 248, 285, 220, 261, 220, 244, 270, 242,
  248, 234, 243, 259, 246, 253, 225, 229, 250, 237, 257, 261, 248,
  219, 244, 283, 293, 312, 272, 280, 298, 275, 297, 350, 260, 313,
  273, 289, 326, 281, 288, 280, 283, 307, 286, 298, 267, 272, 285,
  286, 303, 295, 289, 258, 286, 320, 354, 328, 297, 323, 331, 305,
  338, 376, 296, 352, 314, 325, 358, 312, 324, 316, 317, 336, 321,
  334, 302, 302, 323, 331, 345, 333, 316, 291, 324
), .Dim = c(30, 5))
x <- c(8.0, 15.0, 22.0, 29.0, 36.0)
xbar <- 22.0 

# BUGS rats example (Vol 1, Example 1)
# http://www.openbugs.net/Examples/Rats.html
library(rstan, quietly = TRUE)
rstan_options(auto_write = TRUE)
options(mc.cores = 2)
custom_colors <- c(
  "#4285f4", # GoogleBlue
  "#34A853", # GoogleGreen
  "#FBBC05", # GoogleYellow
  "#EA4335" # GoogleRed
)
rstan_ggtheme_options(
  panel.background = element_rect(fill = "white"),
  legend.position = "top"
)
rstan_gg_options(
  fill = "#4285f4", color = "white",
  pt_color = "#EA4335", chain_colors = custom_colors
)
chains = 4 
iter = 1000 

init <- rep(list(list(
  alpha = rep(250, 30), beta = rep(6, 30),
  alpha_c = 150, beta_c = 10,
  tausq_c = 1, tausq_alpha = 1,
  tausq_beta = 1
)), chains)

rats_fit <- stan(
  model_name = "rats",
  model_code = "
  data {
    int<lower=0> N;
    int<lower=0> T;
    real x[T];
    real y[N,T];
    real xbar;
  }
  parameters {
    real alpha[N];
    real beta[N];
  
    real alpha_c;
    real beta_c;          // beta.c in original bugs model
  
    real<lower=0> tausq_c;
    real<lower=0> tausq_alpha;
    real<lower=0> tausq_beta;
  }
  transformed parameters {
    real<lower=0> tau_c;       // sigma in original bugs model
    real<lower=0> tau_alpha;
    real<lower=0> tau_beta;
  
    tau_c = sqrt(tausq_c);
    tau_alpha = sqrt(tausq_alpha);
    tau_beta = sqrt(tausq_beta);
  }
  model {
    alpha_c ~ normal(0, 100);
    beta_c ~ normal(0, 100);
    tausq_c ~ inv_gamma(0.001, 0.001);
    tausq_alpha ~ inv_gamma(0.001, 0.001);
    tausq_beta ~ inv_gamma(0.001, 0.001);
    alpha ~ normal(alpha_c, tau_alpha); // vectorized
    beta ~ normal(beta_c, tau_beta);  // vectorized
    for (n in 1:N)
      for (t in 1:T) 
        y[n,t] ~ normal(alpha[n] + beta[n] * (x[t] - xbar), tau_c);
  }
  generated quantities {
    real alpha0;
    alpha0 = alpha_c - xbar * beta_c;
  }
  ",
  data = list(N = N, T = T, y = y, x = x, xbar = xbar), 
  chains = chains, init = init, iter = iter)
```

#### Current Output:

My `Makevars.win` file

```
CXX14FLAGS=-O3 -march=native
CXX14 = g++ -m$(WIN) -std=c++1y
CXX11FLAGS=-O3 -march=native
```

and 

```r
pkgbuild::has_build_tools(debug = TRUE)
```
```
Scanning R CMD config CC...
cc_path: c:/Rtools/mingw_64/bin/gcc 
install_path: c:/Rtools 
VERSION.txt
Rtools version 3.5.0.4 
Found compatible gcc from R CMD config CC
[1] TRUE
```

and 

```
fx <- cxxfunction( signature(x = "integer", y = "numeric" ) , '
+ return ScalarReal( INTEGER(x)[0] * REAL(y)[0] ) ;
+ ' )
> fx( 2L, 5 )
[1] 10
```

#### RStan Version:
```
packageVersion("rstan")
[1] ‘2.18.2’
```

#### R Version:
```
R.version.string
[1] "R version 3.6.0 (2019-04-26)"
```

#### Operating System:

Windows 8.1 x64

```
sessionInfo()
R version 3.6.0 (2019-04-26)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 8.1 x64 (build 9600)

Matrix products: default

locale:
[1] LC_COLLATE=Chinese (Simplified)_China.936 
[2] LC_CTYPE=Chinese (Simplified)_China.936   
[3] LC_MONETARY=Chinese (Simplified)_China.936
[4] LC_NUMERIC=C                              
[5] LC_TIME=Chinese (Simplified)_China.936    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] rstan_2.18.2       StanHeaders_2.18.1 ggplot2_3.1.1     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1         magrittr_1.5       tidyselect_0.2.5  
 [4] munsell_0.5.0      colorspace_1.4-1   R6_2.4.0          
 [7] rlang_0.3.4        plyr_1.8.4         dplyr_0.8.0.1     
[10] tools_3.6.0        parallel_3.6.0     pkgbuild_1.0.3    
[13] grid_3.6.0         gtable_0.3.0       loo_2.1.0         
[16] cli_1.1.0          withr_2.1.2        matrixStats_0.54.0
[19] lazyeval_0.2.2     assertthat_0.2.1   tibble_2.1.1      
[22] crayon_1.3.4       processx_3.3.1     gridExtra_2.3     
[25] purrr_0.3.2        callr_3.2.0        ps_1.3.0          
[28] inline_0.3.15      glue_1.3.1         compiler_3.6.0    
[31] pillar_1.4.0       prettyunits_1.0.2  scales_1.0.0      
[34] stats4_3.6.0       pkgconfig_2.0.2   
```

**Comment by nhkim1302** [comment]
Exactly same error occurs to me when running simple rstan example. 
It works well on R 3.6.0 x64, but same error occurs on R 3.6.0 i386.

**Comment by XiangyunHuang** [comment]
@nhkim1302 what's your `Makevars.win` ?

**Comment by ghost** [comment]
I am getting the same error after upgrading `rstan` to version 2.18.2. In an attempt to solve the problem, I uninstalled all things R, then re-installed R (3.6.0) and Rtools, and installed the CRAN versions of `rstan` and dependencies.  Everything is the same as laid out by @XiangyunHuang, except I'm using a Windows 10 operating system and I have Rtools version 3.5.0.1.

**Comment by nhkim1302** [comment]
@XiangyunHuang 
My `Makevars` is as follows:
```
CXX14FLAGS=-O3 -march=native -mtune=native -fPIC
CXX14 = g++ -m$(WIN) -std=c++1y
CXX14=$(BINPREF)g++ -O2 -march=native -mtune=native
```

**Comment by XiangyunHuang** [comment]
Thanks @nhkim1302.

> Exactly same error occurs to me when running simple rstan example.
> It works well on R 3.6.0 x64, but same error occurs on R 3.6.0 i386.

It works on my R 3.6.0 x64 under Windows 8.1 x64, too

and 

```
CXX14=$(BINPREF)g++ -O2 -march=native -mtune=native
```

maybe unnecessary

**Comment by plouis5712** [comment]
I get the same issue with rstan version 2.18.2, Windows 7 OS, R version 3.6.0, and Rtools 3.5.0.4. Here's my session info:
R version 3.6.0 (2019-04-26)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

None of the fixes to the Makevars.win file above eliminate the error for me. Moreover, I had everything working nicely on the pre-2.18 rstan versions (earlier versions of R, Rtools too, but still Windows 7 OS) . Anyone figured this out?

**Comment by bgoodri** [comment]
Install from GitHub

**Comment by bgoodri** [comment]
Specifically, the develop branch

**Comment by plouis5712** [comment]
I’m curious: Do you know which R function is throwing this error? The only rstan package R function I can find calling the “grepl” function is stan_model, and its calls to grepl don’t have the same format reported in the error.

**Comment by bgoodri** [comment]
It is an internal function to rstan called get_makefile_txt or something
like that. Anyway, it is disabled on the develop branch of rstan on GitHub.

**Comment by plouis5712** [comment]
I tried what you suggested—I had to download the development version as a .zip file (from https://github.com/stan-dev/rstan) because curl_fetch_memory kept timing out. Here’s my session info and what happened when I tried installing it:

> sessionInfo()
R version 3.6.0 (2019-04-26)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C
[5] LC_TIME=English_United States.1252

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base

loaded via a namespace (and not attached):
[1] processx_3.3.1    compiler_3.6.0    magrittr_1.5      assertthat_0.2.1  R6_2.4.0          cli_1.1.0
 [7] prettyunits_1.0.2 tools_3.6.0       crayon_1.3.4      callr_3.2.0       ps_1.3.0          pkgbuild_1.0.3


> install.packages("C:\\Stat Software\\rstan-develop\\rstan\\rstan", repos=NULL, dependencies=T, type="src")
* installing *source* package 'rstan' ...
** using staged installation
** libs

*** arch - i386


g++ -m32 -std=c++1y  -I"C:/R/R-36~1.0/include" -DNDEBUG -I"../inst/include" -I"." -I"C:/R/R-3.6.0/library/StanHeaders/include/src" -DBOOST_DISABLE_ASSERTS -DBOOST_PHOENIX_NO_VARIADIC_EXPRESSION -I"C:/R/R-3.6.0/library/Rcpp/include" -I"C:/R/R-3.6.0/library/RcppEigen/include" -I"C:/R/R-3.6.0/library/BH/include" -I"C:/R/R-3.6.0/library/StanHeaders/include"        -O3 -march=native -mtune=native -c sparse_extractors.cpp -o sparse_extractors.o
In file included from C:/Rtools/mingw_64/x86_64-w64-mingw32/include/c++/random:49:0,
                 from C:/Rtools/mingw_64/x86_64-w64-mingw32/include/c++/bits/stl_algo.h:66,
                 from C:/Rtools/mingw_64/x86_64-w64-mingw32/include/c++/algorithm:62,
                 from C:/R/R-3.6.0/library/RcppEigen/include/Eigen/Core:288,
                 from C:/R/R-3.6.0/library/RcppEigen/include/Eigen/Dense:1,
                 from C:/R/R-3.6.0/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:4,
                 from C:/R/R-3.6.0/library/StanHeaders/include/stan/math/prim/mat/fun/csr_extract_u.hpp:5,
                 from sparse_extractors.cpp:2:
C:/Rtools/mingw_64/x86_64-w64-mingw32/include/c++/bits/random.h:106:26: error: expected unqualified-id before '__int128'
       { typedef unsigned __int128 type; };
                          ^
make: *** [C:/R/R-36~1.0/etc/i386/Makeconf:215: sparse_extractors.o] Error 1
ERROR: compilation failed for package 'rstan'
* removing 'C:/R/R-3.6.0/library/rstan'
Warning message:
In install.packages("C:\\Stat Software\\rstan-develop\\rstan\\rstan",  :
  installation of package ‘C:/STATSO~1/RSTAN-~1/rstan/rstan’ had non-zero exit status

**Comment by plouis5712** [comment]
I think this is Windows OS version related. I just installed rstan at home on my Windows 10 machine using Rtools version 3.5.0.4 and R version 3.6.0 (2019-04-26). I followed the instructions at https://github.com/stan-dev/rstan/wiki/RStan-Getting-Started; in particular, the Makevars.win file only had 2 lines:
CXX14FLAGS=-O3 -march=native -mtune=native

CXX11FLAGS=-O3 -march=native -mtune=native

And then I used install.packages(“rstan”, dependencies=T) with default repository http://cran.cnr.berkeley.edu/. The 8 schools example ran without any problem. At work I have Windows 7 x64 (build 7601) Service Pack 1—this is what’s been giving all the errors.
